core/src/main/scala/org/apache/spark/crypto/CryptoStreamUtils.scala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import java.io.{IOException, InputStream, OutputStream}
+import java.nio.ByteBuffer
+import java.util.Properties
+
+import com.intel.chimera._
+import com.intel.chimera.utils.Utils
+import sun.nio.ch.DirectBuffer
+
+import org.apache.spark.SparkConf
+import org.apache.spark.crypto.CommonConfigurationKeys._
+import org.apache.spark.deploy.SparkHadoopUtil
+
+/**
+ * A util class for [[CryptoInputStream]] and [[CryptoOutputStream]]
+ */
+private[spark] object CryptoStreamUtils {
+  val IV_LENGTH = 16
+  val SPARK_CHIMERA_CONF_PREFIX = "spark.shuffle.chimera"
```

Comment:
These should end with `.`, otherwise the code in `toChimeraConf` will translate `spark.shuffle.chimeraFoo.bar` to `chimeraFoo.bar`.